src/crimson/os/seastore/cache.h

```diff
@@ -731,6 +728,11 @@ class Cache {
     }
   }
 
+  struct src_ext_t {
+    Transaction::src_t tsrc = Transaction::src_t::MAX;
+    uint32_t fadvise_flags = 0;
```

Comment:
I think this is not appropriate for 2 reasons:
1. `src_ext_t` wasn't introduced for this purpose, `src_t` was intended only be used for statistics/metrics purposes, not for advising the read/write behaviors.
2. `fadvise_flags` is OSD's semantic, not SeaStore -- not all flags are related/implemented from SeaStore perspective.

Like the current `placement_hint_t` (HOT/COLD/REWRITE) which is used to advise writes, probably there is a need to introduce a enum `read_hint_t` (HOT/COLD) to guide read behaviors, probably to start at `Cache::get_xxx_extent_xxx` public interfaces.

In the future, I suppose it might be able to be extended to be combined with (Sequential/Random etc.) hints to guide prefetching as well.

Reply:
I think you mean that we should define dedicated hint for SeaStore to avoid the cache pollution instead of reusing OSDs flags, am I right?

Reply:
> 1. `src_ext_t` wasn't introduced for this purpose, `src_t` was intended only be used for statistics/metrics purposes, not for advising the read/write behaviors.

It seems that `Cache::LRU::move_to_top()` relies on `src_t` to determine whether the extent should be touched.

Reply:
IIUC, we should also modify `Cache::LRU::move_to_top()` to rely on hints instead of `src_t`, am I right?

Reply:
How is `src_t` in `LRU::move_to_top()` related?

It seems to me that `src_t` in `touch_extent()` is related -- I suppose it is simple to be converted once `read_hint_t` is introduced.

Reply:
> How is `src_t` in `LRU::move_to_top()` related?
> 
> It seems to me that `src_t` in `touch_extent()` is related -- I suppose it is simple to be converted once `read_hint_t` is introduced.

Ah, yes, it's `touch_extent()`, sorry for the mistake. Will modify this PR:-)